### PR TITLE
add eager completion option for zipLatest

### PR DESCRIPTION
**Pull request by contrun**
Reference: https://github.com/akka/akka/issues/25839

### CODE REVIEW

**Review by ennru — APPROVED** [review]
Sorry for not getting to this earlier.
LGTM.

**Comment by ennru** [comment]
But Mima is not that happy about the new parameter: https://travis-ci.org/github/akka/akka/jobs/685257534#L308

**Comment by contrun** [comment]
@ennru Yeah. I saw the problem, but I didn't come up with a solution yet. I don't know how to circumvent it.

**Review by ennru — CHANGES_REQUESTED** [review]
Mima complains for good reasons. I added the old constructor in the generator pattern.
Avoid default values to parameters in APIs that intend to stay backwards compatible.
To make this work, you'll need a new method name in `FlowOps` to disambiguate the methods.

This PR lacks a Java DSL counterpart.

**Comment by hepin1989** [comment]
@raboof @johanandren hi team, @contrun have updated this PR, any inputs?

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 1]
```diff
@@ -2779,17 +2779,43 @@ trait FlowOps[+Out, +Mat] {
    *
    *   '''Backpressures when''' downstream backpressures
    *
-   *   '''Completes when''' any of the upstreams completes
+   *   '''Completes when''' any upstream completes if `eagerComplete` is enabled or wait for all upstreams to complete
```
Comment (ennru) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala:2782]: This method signature doesn't have the `eagerComplete` parameter.
```suggestion
   *   '''Completes when''' any of the upstreams completes
```

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 2]
```diff
@@ -2779,17 +2779,43 @@ trait FlowOps[+Out, +Mat] {
...
+   * No element is emitted until at least one element from each Source becomes available. Whenever a new
+   * element appears, the zipping function is invoked with a tuple containing the new element
+   * and the other last seen elements.
+   *
+   *   '''Emits when''' all of the inputs have at least an element available, and then each time an element becomes
+   *   available on either of the inputs
+   *
+   *   '''Backpressures when''' downstream backpressures
+   *
+   *   '''Completes when''' any of the upstreams completes
```
Comment (ennru) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala:2802]:
```suggestion
   *   '''Completes when''' any upstream completes if `eagerComplete` is enabled or wait for all upstreams to complete
```

File: akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala [hunk 3]
```diff
@@ -2779,17 +2779,43 @@ trait FlowOps[+Out, +Mat] {
...
+   *
+   *   '''Cancels when''' downstream cancels
+   */
+  def zipLatestWith[Out2, Out3](that: Graph[SourceShape[Out2], _], eagerComplete: Boolean)(
+      combine: (Out, Out2) => Out3): Repr[Out3] =
+    via(zipLatestWithGraph(that, eagerComplete)(combine))
+
+  //TODO exclude this in mima ,this method is not used.
```
Comment (ennru) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala:2810]:
```suggestion
```

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphZipLatestWithSpec.scala [hunk 4]
```diff
@@ -26,8 +26,15 @@ class GraphZipLatestWithSpec extends TwoStreamsSetup {
   }
 
   override def setup(p1: Publisher[Int], p2: Publisher[Int]) = {
+    setup(p1, p2, eagerComplete = false)
```
Comment (ennru) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/GraphZipLatestWithSpec.scala:29]: Eager completion should stay to be the default, correct?

File: akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala [hunk 5]
```diff
@@ -1033,10 +1033,12 @@ final class Zip[A, B] extends ZipWith2[A, B, (A, B)](Tuple2.apply) {
 
 object ZipLatest {
 
+  def apply[A, B](): ZipLatest[A, B] = new ZipLatest()
```
Comment (ennru) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala:1036]: Doc comment missing.

File: akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala [hunk 6]
```diff
@@ -1100,7 +1109,7 @@ object ZipLatestWith extends ZipLatestWithApply
  *
  * '''Backpressures when''' any of the outputs backpressure
  *
- * '''Completes when''' upstream completes
+ * '''Completes when''' any upstream completes if `eagerComplete` is enabled or wait for all upstreams to complete
```
Comment (ennru) [inline akka-stream/src/main/scala/akka/stream/scaladsl/Graph.scala:1112]: Unzip doesn't change.

**Review by ennru — APPROVED** [review]
LGTM.

**Review by johanandren — APPROVED** [review]
LGTM, thanks!

**Pull request merged**